src/include/duckdb/common/chrono.hpp

```diff
@@ -17,4 +17,9 @@ using std::chrono::high_resolution_clock;
 using std::chrono::milliseconds;
 using std::chrono::system_clock;
 using std::chrono::time_point;
+
+// The portable way to do this (steady_clock) is not performant on all platforms.
```

Comment:
Could you expand on what platforms this isn't performant, and why it isn't performant? Could we use `steady_clock::now()` on platforms where it is performant?

Reply:
My initial implementation used steady_clock because initial research suggested that it should be broadly performant. However, when I submitted the PR, there was an unmistakable perf regression reported by CI. I switched to this implementation and the regression went away.

Unfortunately, since steady_clock makes no actual guarantees about performance, it seemed safer to provide an implementation that uses the most performant method on each platform.

Reply:
It looks like github may have purged the runs from the original commit on this PR, which used steady_clock.

I created a new PR that uses steady clock, for perf comparison. https://github.com/duckdb/duckdb/pull/11904 If that one isn't worse, I will clean up this PR to use steady_clock.

Reply:
That other PR didn't complain about perf. Should I switch this back to steady_clock?

Reply:
Yes, let's switch it to steady_clock, we can always switch back